src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1,25 +1,633 @@
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/func_80800B90.s")
+/*
+ * File: z_select.c
+ * Overlay: ovl_select
+ * Description: Debug Scene Select Menu
+ */
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/func_80800BAC.s")
+#include <ultra64.h>
+#include <global.h>
+#include <vt.h>
+#include <alloca.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/func_80800CB4.s")
+void Select_LoadTitle(SelectContext* this) {
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, Title_Init, TitleContext);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/func_80801614.s")
+void Select_LoadGame(SelectContext* this, s32 entranceIndex) {
+    osSyncPrintf(VT_FGCOL(BLUE));
+    osSyncPrintf("\n\n\nＦＩＬＥ＿ＮＯ＝%x\n\n\n", gSaveContext.fileNum);
+    osSyncPrintf(VT_RST);
+    if (gSaveContext.fileNum == 0xFF) {
+        func_800A82C8();
+        gSaveContext.unk_13F6 = gSaveContext.magic;
+        gSaveContext.magic = 0;
+        gSaveContext.unk_13F4 = 0;
+        gSaveContext.magicLevel = gSaveContext.magic;
+    }
+    gSaveContext.buttonStatus[4] = BTN_ENABLED;
+    gSaveContext.buttonStatus[3] = BTN_ENABLED;
+    gSaveContext.buttonStatus[2] = BTN_ENABLED;
+    gSaveContext.buttonStatus[1] = BTN_ENABLED;
+    gSaveContext.buttonStatus[0] = BTN_ENABLED;
+    gSaveContext.unk_13E7 = gSaveContext.unk_13E8 = gSaveContext.unk_13EA = gSaveContext.unk_13EC = 0;
+    Audio_SetBGM(NA_BGM_STOP);
+    gSaveContext.entranceIndex = entranceIndex;
+    gSaveContext.respawnFlag = 0;
+    gSaveContext.respawn[0].entranceIndex = -1;
+    gSaveContext.seqIndex = 0xFF;
+    gSaveContext.nightSeqIndex = 0xFF;
+    gSaveContext.unk_13C7 = 1;
+    D_8011FB30 = 0;
+    this->state.running = false;
+    SET_NEXT_GAMESTATE(&this->state, Gameplay_Init, GlobalContext)
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/func_808017F0.s")
+static SceneSelectEntry sScenes[] = {
+    { " 1:SPOT00", Select_LoadGame, 0x000000CD },
```

Comment:
Entrance indexes are normally written as shorts, not words.
```suggestion
    { " 1:SPOT00", Select_LoadGame, 0x00CD },
```